src/main/java/com/lpvs/service/scan/LPVSDetectService.java

```diff
@@ -105,52 +111,118 @@ public LPVSDetectService(
     public void runSingleScan() {
-        if (trigger != null && !HtmlUtils.htmlEscape(trigger).isEmpty()) {
-            log.info("Triggered single scan operation");
+        // generateReport indicates that a report should be generated (HTML or command line output)
+        boolean generateReport = false;
+        LPVSQueue webhookConfig = null;
+        List<LPVSFile> scanResult = null;
+        List<LPVSLicenseService.Conflict<String, String>> detectedConflicts = null;
+
+        // Error case when both pull request scan and local files scan are set to true
+        if (trigger != null && !trigger.isBlank() && localPath != null && !localPath.isBlank()) {
+            log.error(
+                    "Incorrect settings: both pull request scan and local files scan are set to true.");
+            SpringApplication.exit(ctx, () -> 0);
+
+            // Scan option - single pull request scan
+        } else if (trigger != null && !HtmlUtils.htmlEscape(trigger).isBlank()) {
+            log.info("Triggered single scan of pull request.");
             try {
                 licenseService.reloadFromTables();
-                LPVSQueue webhookConfig =
+                webhookConfig =
                         gitHubService.getInternalQueueByPullRequest(HtmlUtils.htmlEscape(trigger));
 
-                List<LPVSFile> scanResult =
+                scanResult =
                         this.runScan(
                                 webhookConfig, gitHubService.getPullRequestFiles(webhookConfig));
 
-                List<LPVSLicenseService.Conflict<String, String>> detectedConflicts =
-                        licenseService.findConflicts(webhookConfig, scanResult);
-
-                if (htmlReport != null && !HtmlUtils.htmlEscape(htmlReport).isEmpty()) {
-                    Path buildReportPath = Paths.get(htmlReport);
-                    Path parentDirectory = buildReportPath.getParent();
-
-                    if (parentDirectory != null && Files.isDirectory(parentDirectory)) {
-                        String report =
-                                LPVSCommentUtil.buildHTMLComment(
-                                        webhookConfig, scanResult, detectedConflicts);
-                        LPVSCommentUtil.saveHTMLToFile(report, buildReportPath.toString());
-                    } else {
-                        log.error(
-                                "Error: The parent directory '"
-                                        + parentDirectory
-                                        + "' does not exist.");
-                    }
+                detectedConflicts = licenseService.findConflicts(webhookConfig, scanResult);
+                generateReport = true;
+                log.info("Single scan of pull request completed.");
+            } catch (Exception ex) {
+                log.error("Single scan of pull request finished with errors.");
+                log.error("Can't trigger single scan: " + ex.getMessage());
```

Comment:
"Single scan finished" and at the same time "Can't trigger single scan"
What about use something general: ""
`log.error("Single scan of pull request failed with error: " + ex.getMessage());`